www/js/goals.js

```diff
@@ -61,37 +81,340 @@ angular.module('emission.main.goals',['emission.services', 'ngSanitize'])
 		});
 		CommHelper.habiticaRegister(regConfig, function(response) {
 			console.log("Success!")
-			$scope.screen = response.success;
-			console.log(response);
-			$window.location.reload();
+			getUserInfo();
 		}, function(error) {
 			$ionicLoading.hide();
 			$ionicPopup.alert({title: "<h4 class='center-align'>Username is Required</h4>",
 								okText: "Try Again",
 								okType: "button-assertive"});
-			console.log(JSON.stringify(error));
+			console.log("Not signed up");
 		});
 	};
  
+
     $ionicLoading.show({
 			template: '<ion-spinner icon="bubbles" class="costume"></ion-spinner>'
+	});
+
+	var getUserInfo = function(){
+		var callOpts = {'method': 'GET', 'method_url': "/api/v3/user",
+	                    'method_args': null};
+
+		CommHelper.habiticaProxy(callOpts, function(response){
+			$scope.screen = response.success;
```

Comment:
I am not sure that this works the way that you want. Note that the `habiticaProxy` method is asynchronous. Let us assume that you invoke it when the controller is instantiated. It is not going to be true immediately - in particular, when the HTML is loading and the value of the variable is evaluted in `ng-if`.

Then, when the async call does complete and you set it, angular will not re-read the variable unless you indicate that it has changed. You are not doing `$scope.$apply()` so I don't see why that would happen.

Are you sure that this works? Not just by reloading from safari, but from actually relaunching the app in the emulator.